components/brave_wallet_ui/page/screens/composer_ui/token_list_item/token_list_item.tsx

```diff
@@ -42,21 +40,30 @@ import {
   IconAndName,
   NameAndBalanceText,
   NetworkAndFiatText,
-  DisabledLabel
+  DisabledLabel,
+  PercentChangeText,
+  PercentChangeIcon,
+  AccountsIcon,
+  InfoButton,
+  InfoIcon
 } from './token_list_item.style'
 import {
   Column,
   Row,
-  HorizontalSpace
+  HorizontalSpace,
+  VerticalSpace,
+  Text
 } from '../../../../components/shared/style'
 
 interface Props {
   onClick: () => void
-  token: BraveWallet.BlockchainToken
-  balance?: string
-  spotPrice?: string
+  onViewTokenDetails: (token: BraveWallet.BlockchainToken) => void
+  token: UserAssetInfoType
```

Comment:
can we keep this as seperate token & balance props?

Reply:
Wouldn't we want to reduce `props` if the balance is available in `UserAssetInfoType`?

Reply:
The issue I have with the `UserAssetInfoType` type is that if a single token's balance changes, the entire list of tokens+balances is invalidated, causing the entire list to re-render. if the balances are tracked separately, balances can be updated in a way that only the affected item will need to re-render

Reply:
I see! Okay this has been updated to use a `memoized` balance registry!